Tools used in this manuscript
----------------------------------------------------------------------

RNA-seq:
* [HISAT2](http://daehwankimlab.github.io/hisat2/)<sup>1</sup> to align reads from RNA-seq experiments
* [Rsubread](https://bioconductor.org/packages/release/bioc/html/Rsubread.html)<sup>2</sup> to count transcripts
* [DESeq2](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)<sup>2</sup> for detection of differentially expressed genes
* [Morpheus](https://software.broadinstitute.org/morpheus/) to visualize count data as an interactive heatmap online. Good for quickly visualizing counts for genes of interest (webtool!)
* [pheatmap](https://cran.r-project.org/package=pheatmap) and [ComplexHeatmap](https://jokergoo.github.io/ComplexHeatmap-reference/book/index.html)<sup>2</sup> to generate heatmaps of the count matrices
* [ggVolcanoR](https://ggvolcanor.erc.monash.edu/) to create Volcano Plots of RNA-seq data (webtool!)
* [ShinyGO](http://bioinformatics.sdstate.edu/go/) for gene-set enrichment analysis of differentially expressed genes from RNA-seq and generation of lollipop plots (webtool!)
* [BioTapestry](https://biotapestry.systemsbiology.net/) to build and visualize the gene regulatory network models

ChIP-seq:
* [ENCODE Transcription Factor and Histone ChIP-Seq processing pipeline](https://github.com/ENCODE-DCC/chip-seq-pipeline2)<sup>1</sup> to process ChIP-seq replicates and identify highly reproducible peaks. 
  *  See [this](https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit) document for more details on the pipeline and [this](https://sites.google.com/site/anshulkundaje/projects/idr) website.
  *  [Here](https://hbctraining.github.io/Intro-to-ChIPseq/lessons/07_handling-replicates-idr.html) is a great writeup on how the Irreproducibility Discovery Rate (IDR) analysis is used to combine replicates and identify highly reproducible peaks.
* [Hierarchical Alignment (HAL)](https://github.com/ComparativeGenomicsToolkit/hal) (LiftOver)<sup>1</sup> to map ChIP-seq peak coordinates from reptile genomes to the mouse genome using a multi-species alignment generated by Brant Faircloth.
* [GREAT (Genomic Regions Enrichment of Annotations Tool)](http://great.stanford.edu/public/html/) to assign putative target genes with each mouse peak (webtool!)
* [HOMER (Hypergeometric Optimization of Motif EnRichment)](http://homer.ucsd.edu/homer/)<sup>1</sup> to perform motif analyses (findMotifsGenome.pl)
* [deepTools](https://deeptools.readthedocs.io/en/develop/)<sup>3</sup> to generate enrichment heatmaps
* [pyGenomeTracks](https://pygenometracks.readthedocs.io/en/latest/)<sup>3</sup> to visualize genome browser tracks of peaks of interest
* [bedtools](https://bedtools.readthedocs.io/en/latest/index.html)<sup>1</sup> to find peak coordinate intersections
* [Canva](https://www.canva.com/graphs/doughnut-charts/) to generate doughnut charts (webtool!) - Final edits made in Illustrator

For ChIP-seq analyses in which only one replicate was generated:
* [bwa]()<sup>1</sup> to align reads
* [MACS2]()<sup>1</sup> to call peaks


***Notes:***
* <sup>1</sup>  ran on the Sapelo2 cluster of the Georgia Advanced Computing Resource Center ([GACRC](https://gacrc.uga.edu/))
* <sup>2</sup> ran in R/ Rstudio
* <sup>3</sup>  ran locally at the command line within [Linux Mint](https://www.linuxmint.com/) running in a [Virtual Machine](https://www.virtualbox.org/)

After generation of figures, vector graphics were edited in Adobe Illustrator and assembled for publication in Adobe InDesign. Other photo edits were made in either Adobe Photoshop or Lightroom.

  
